purchase_order_supplierinfo_update/models/purchase_order.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("state", "") in ["purchase", "done"]:
+            self.mapped("order_line").update_supplierinfo_prince()
+        return res
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res.update_supplierinfo_prince()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("price_unit"):
+            self.update_supplierinfo_prince()
+        return res
+
+    def update_supplierinfo_prince(self):
+        for line in self.filtered(lambda r: r.order_id.state in ["purchase", "done"]):
+            domain = [
+                ("partner_id", "=", line.partner_id.id),
+                ("product_id", "=", line.product_id.id),
+                ("date_order", ">", line.date_order),
+            ]
+            if not self.env["purchase.order.line"].search(domain, limit=1):
+                # The same domain used in purchase.order.line to suggest the price.
+                # see _suggest_quantity method in models/purchase.py of odoo core
+                # purchase module.
+                supplierinfo = self.product_id.seller_ids.filtered(
+                    lambda r: (
+                        r.name == self.order_id.partner_id
+                        and (not r.product_id or r.product_id == self.product_id)
+                    )
+                ).sorted(key=lambda r: r.min_qty)
+                if supplierinfo and line.price_unit != supplierinfo[0].price:
```

Comment:
Please create a function for that. ``def _update_supplierinfo(self, line)``.

that will allow inheritance. otherwise, it's not possible to make glue modules with purchase_discount and purchase_triple_discount modules.

Reply:
@legalsylvain I have extract the supplierinfo update in a separate method to make it extensible as you suggested, could you check it, please?